applications/OptimizationApplication/python_scripts/utilities/opt_line_search.py

```diff
@@ -144,15 +144,19 @@ def ComputeStep(self) -> KratosOA.CollectiveExpression:
             d = algorithm_buffered_data.GetValue("control_field_update", 1)
             d = d.Evaluate()
             for i in range(len(y)):
-                dy = d[i] * y[i]
-                dd = d[i] * d[i]
+                dy = numpy.dot(d[i], y[i])
+                dd = numpy.dot(d[i], d[i])
+                
                 if math.isclose(dy, 0.0, abs_tol=1e-16):
                     self.step_numpy[i] = self._max_step / norm
                 else:
                     self.step_numpy[i] = abs( dd / dy )
-                if self.step_numpy[i] > self._max_step / norm:
-                    self.step_numpy[i] = self._max_step / norm
 
+                if isinstance(d[i], (float, int, numpy.float64)) and self.step_numpy[i] > self._max_step / norm:
+                    self.step_numpy[i] = self._max_step / norm
+                elif isinstance(d[i], (numpy.ndarray)) and self.step_numpy[i][0] > self._max_step / norm:
```

Comment:
Woudn't it be better to use a norm like L2 in here instead of checking the first entry?